### PR TITLE
Fix crash in SelectNextItem when list is empty or no item selected

### PR DESCRIPTION
**Pull request by Don-Vito**
## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/issues/8421
* [x] CLA signed. 
* [x] Tests added/passed

## Detailed Description of the Pull Request / Additional comments
Skip further computations if:
* The list is empty
* No item is selected and we press "up" key 
(we do want to handle "down" key for CommandLineMode)

## Validation Steps Performed
* Manual testing + UT

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]

File: src/cascadia/TerminalApp/CommandPalette.cpp [hunk 1]
```diff
@@ -110,12 +110,19 @@ namespace winrt::TerminalApp::implementation
...
-        // Wraparound math. By adding numItems and then calculating modulo numItems,
-        // we clamp the values to the range [0, numItems) while still supporting moving
-        // upward from 0 to numItems - 1.
-        const auto newIndex = ((numItems + selected + (moveDown ? 1 : -1)) % numItems);
-        _filteredActionsView().SelectedIndex(newIndex);
-        _filteredActionsView().ScrollIntoView(_filteredActionsView().SelectedItem());
+
+        // Do not try to select an item if
+        // - the list is empty
+        // - if no item is selected and "up" is pressed
```
Comment (rheh) [inline src/cascadia/TerminalApp/CommandPalette.cpp:116]: Question: the comment refers to 'up', but the if statement moveDown.  What am I missing?

File: src/cascadia/TerminalApp/CommandPalette.cpp [hunk 2]
```diff
@@ -110,12 +110,19 @@ namespace winrt::TerminalApp::implementation
...
+        if (numItems != 0 && (selected != -1 || moveDown))
+        {
+            // Wraparound math. By adding numItems and then calculating modulo numItems,
+            // we clamp the values to the range [0, numItems) while still supporting moving
+            // upward from 0 to numItems - 1.
+            const auto newIndex = ((numItems + selected + (moveDown ? 1 : -1)) % numItems);
```
Comment (rheh) [inline src/cascadia/TerminalApp/CommandPalette.cpp:122]: nit pick: outer braces not required.

File: src/cascadia/TerminalApp/CommandPalette.cpp (hunk 1 above)
Comment (Don-Vito) [reply src/cascadia/TerminalApp/CommandPalette.cpp:116]: Upon down we go to 0, which is a desired behavior in CommandLine mode. I agree that it is not very explicit and probably worth commenting

File: src/cascadia/TerminalApp/CommandPalette.cpp (hunk 2 above)
Comment (Don-Vito) [reply src/cascadia/TerminalApp/CommandPalette.cpp:122]: I agree - but this is and old code - I just added the condition.

File: src/cascadia/TerminalApp/CommandPalette.cpp (hunk 1 above)
Comment (Don-Vito) [reply src/cascadia/TerminalApp/CommandPalette.cpp:116]: Sorry - for confusing comment
* In the comment: don't try to select if up is pressed
* In the code: try to select if down is pressed

!moveDown = up

**Review by rheh — APPROVED** [review]
LGMT

**Review by PankajBhojwani — APPROVED** [review]

**Pull request merged**